crates/router/src/core/payments/gateway/context.rs

```diff
@@ -66,6 +75,25 @@ impl RouterGatewayContext {
             creds_identifier,
         }
     }
+    pub fn default(
+        platform: Platform,
+        #[cfg(feature = "v1")] merchant_connector_account: helpers::MerchantConnectorAccountType,
+        #[cfg(feature = "v2")]
+        merchant_connector_account: hyperswitch_domain_models::merchant_connector_account::MerchantConnectorAccountTypeDetails,
+        merchant_id: id_type::MerchantId,
+        profile_id: id_type::ProfileId,
+    ) -> Self {
+        let lineage_ids = LineageIds::new(merchant_id, profile_id);
+        Self {
+            platform,
+            header_payload: HeaderPayload::default(),
+            lineage_ids,
+            merchant_connector_account,
+            execution_mode: ExecutionMode::NotApplicable,
+            execution_path: ExecutionPath::Direct,
+            creds_identifier: None,
```

Comment:
This is hardcoded here, if the parent function has creds and if the direct/default function is used means we will miss the creds in flow, so lets not hardcode here instead accept the value as argument